OSD Hud + LoRa radar (nearby aircrafts)

OlivierC-FR opened the pull request:
As a subset of the crosshair, adds the ability for the OSD to display points of interest (POI) in "3D".
Adds the support to display the nearby aircrafts from the project iNav Radar, using a 433/868/915MHz ESP32 modem, see here : https://github.com/mistyk/inavradar-ESP32

Samples : 
https://drive.google.com/file/d/1fiEglzfM1DVzJaWzt7Yn5e-RmmaHZMy0/view (some  visuals are outdated)
Pursuit : https://www.youtube.com/watch?v=7ww0YOGN7F0

------------

This is the 3D mode, point A is my flying wing broadcasting position with the 433MHz modem.
4 bars means the link quality is 100%, and distance 10 meters.
H is the home point, 90° to the left, at 3 meters.
 
![000](https://user-images.githubusercontent.com/25566932/55166582-ae5c1200-516f-11e9-8a0c-16f262846e85.jpg)

Map mode, view from above. Center arrow is you, forward is the top, and the A is the other aircraft.

![001](https://user-images.githubusercontent.com/25566932/55166633-c2077880-516f-11e9-9871-6a77e65781df.jpg)

OlivierC-FR committed:
Hud + radar

Rebased

OlivierC-FR committed:
Fixing few inaccuracies in the comments

OlivierC-FR committed:
Home point not displayed 3D by default

OlivierC-FR committed:
Only 5 POIs (Hud display cache)

DzikuVx commented on src/main/io/osd.c:
```diff
@@ -190,6 +182,8 @@ static displayPort_t *osdDisplayPort;
 
 PG_REGISTER_WITH_RESET_FN(osdConfig_t, osdConfig, PG_OSD_CONFIG, 7);
```
@OlivierC-FR you have to bump PG version to avoid clashes during migration without full chip erase

OlivierC-FR replied:
done

DzikuVx commented on src/main/io/osd.c:
```diff
@@ -190,6 +182,8 @@ static displayPort_t *osdDisplayPort;
 
 PG_REGISTER_WITH_RESET_FN(osdConfig_t, osdConfig, PG_OSD_CONFIG, 7);
 
+
```
I know, I'm just just making this stupid comment about empty lines

OlivierC-FR replied:
done

DzikuVx commented on src/main/io/osd_hud.c:
```diff
@@ -0,0 +1,356 @@
+/*
+ * This file is part of Cleanflight.
```
Let's change it to INAV

OlivierC-FR replied:
done

DzikuVx commented on src/main/msp/msp_protocol.h:
```diff
@@ -304,6 +304,10 @@
 // #define MSP_ALARMS               242
 
+// radar commands
+#define MSP_SET_RADAR_POS        248 //SET radar position information
```
I would prefer to have it in MSP2 range but OK, probably it's more probable to get 3rd party support if MSPv1 is used since there are libs that work with MSPv1

OlivierC-FR replied:
Yes I think MSP1 will make it compatible with a lot more other projects

OlivierC-FR committed:
Per review

shellixyz committed:
Switch RADAR MSP messages to MSPv2

shellixyz committed:
Move distance conversion macros to common/maths.h

shellixyz committed:
A lot of stuff fixed

OlivierC-FR committed:
Make distance overwrite existing chars in the Hud

... before it was "behind"

OlivierC-FR committed:
Merge remote-tracking branch 'origin/oc_hud' into oc_hud

OlivierC-FR committed:
Fixes

OlivierC-FR committed:
Relative altitude computation moved from nav to OSD + fix + 5 nodes max

OlivierC-FR committed:
Radar relative positions now computed right before display

... instead of in navigation / at MSP2 receive, this greatly improves the 3D tracking of the point of interest

shellixyz committed:
Merge remote-tracking branch 'upstream/development' into oc_hud

shellixyz committed:
Move constants to common/constants.h

OlivierC-FR committed:
Nearest displayed POI  made smarter and less intrusive

OlivierC-FR committed:
Nearest POI datas display line moved 1 line down

... so it does not collide with the 3D tracking

OlivierC-FR committed:
OSD reordering

OlivierC-FR committed:
Cosmetic tweaks to avoid OSD items collisions

shellixyz committed:
Merge remote-tracking branch 'upstream/development' into oc_hud

OlivierC-FR committed:
Extra POI datas off by default
